src/main/java/sparksoniq/jsoniq/runtime/iterator/control/SwitchRuntimeIterator.java

```diff
@@ -89,38 +89,38 @@ private void initializeIterator(
             Map<RuntimeIterator, RuntimeIterator> cases,
             RuntimeIterator defaultReturn
     ) {
-        Item testValue = getSingleItemFromIterator(
-            test
-        );
-
-        if (testValue != null && testValue.isArray()) {
-            throw new NonAtomicKeyException(
-                    "Invalid args. Switch condition can't be an array type",
-                    getMetadata().getExpressionMetadata()
-            );
-        } else if (testValue != null && testValue.isObject()) {
-            throw new NonAtomicKeyException(
-                    "Invalid args. Switch condition  can't be an object type",
-                    getMetadata().getExpressionMetadata()
-            );
-        }
```

Comment:
testValue != null checks are unified in an outer if for both testValue and caseValue variables